### PR TITLE
ESLint: Add rule to check for unnecessary boxShadow

### PR DESCRIPTION
**Pull request by ayeshakmaz**
Check for any boxShadows set on Box via dangerouslySetInlineStyle that could use borderStyle="shadow" instead

## Test Plan

Added test

### CODE REVIEW

File: packages/eslint-plugin-gestalt/src/no-dangerous-style-duplicates.test.js [hunk 1]
```diff
@@ -389,5 +396,16 @@ ruleTester.run('no-dangerous-style-duplicates', rule, {
         },
       ],
     },
+    {
+      code: invalidBoxShadow,
+      parserOptions,
+      errors: [
+        {
+          message:
+            'Un-needed Box dangerous styles found. https://gestalt.netlify.app/gestalt/#/Box\n' +
```
Comment (dangerismycat) [inline packages/eslint-plugin-gestalt/src/no-dangerous-style-duplicates.test.js:405]: I think that URL needs to be updated to remove the `/gestalt/#`

Comment (dangerismycat) [reply packages/eslint-plugin-gestalt/src/no-dangerous-style-duplicates.test.js:405]: Actually, might as well go right to the business: https://gestalt.netlify.app/Box#Borders

File: packages/eslint-plugin-gestalt/src/validators.js [hunk 2]
```diff
@@ -109,14 +109,47 @@ export const validateBorder = (value: string): ?string => {
...
+};
+
+export const validateBoxShadow = (value: string): ?string => {
+  // If the value is a string:
+  // 1) strip out the rgba portion
+  // 2) convert the pixel portion to only numbers
+  // 3) If both pieces match, recommend borderStyle="shadow"
+
+  const rgbaRegex =
+    '/rgba\\(\\s*(-?\\d+|-?\\d*\\.\\d+(?=%))(%?)\\s*,\\s*(-?\\d+|-?\\d*\\.\\d+(?=%))(\\2)\\s*,\\s*(-?\\d+|-?\\d*\\.\\d+(?=%))(\\2)\\s*,\\s*(-?\\d+|-?\\d*.\\d+)\\s*\\)/g';
```
Comment (dangerismycat) [inline packages/eslint-plugin-gestalt/src/validators.js:131]: :dragon-faint:

File: packages/eslint-plugin-gestalt/src/validators.js [hunk 3]
```diff
@@ -109,14 +109,47 @@ export const validateBorder = (value: string): ?string => {
...
+
+  let rgbaMatch = false;
+  let pixelsMatch = false;
+  if (
+    cleanRgbaPortion &&
+    (cleanRgbaPortion === 'rgba(0,0,0,0.1)' || cleanRgbaPortion === 'rgba(0,0,0,.1)')
+  ) {
+    rgbaMatch = true;
+  }
+  if (cleanPixelPortion === '008' || cleanPixelPortion === '0080') {
```
Comment (dangerismycat) [inline packages/eslint-plugin-gestalt/src/validators.js:147]: nit: I'm a fan of inverting multiple checks of the same variable to something like `['008', '0080'].includes(cleanPixelPortion)`. It's a toss-up on two comparisons, but helps with readability with 3+

**Review by dangerismycat — APPROVED** [review]
Awesome!

File: packages/eslint-plugin-gestalt/src/validators.js (hunk 2 above)
Comment (ayeshakmaz) [reply packages/eslint-plugin-gestalt/src/validators.js]: Yeah I definitely did not make this 😂 thank you internet

**Pull request merged**